Отправка на ревью 7 спринта (первая попытка)

ArtemixSled opened the pull request:
Реализован класс FileBackedTaskManager + JUnit тесты к нему

ArtemixSled committed:
Отправка на ревью 7 спринта (первая попытка)

AlexeyCheuzov commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,193 @@
+        save();
+    }
+
+    static class ManagerSaveException extends RuntimeException {
```
Ошибку стоит вынести в отдельный класс

AlexeyCheuzov commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,193 @@
+                Task task = fromString(fileReader.readLine());
+
+                if (task instanceof Epic) {
+                    fileBackedTaskManager.createEpic((Epic) task);
```
Не надо использовать методы по созданию задач, они вызывают save. Хранилища в родительском класс можно сделать protected и добавлять задачи напрямую

AlexeyCheuzov commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,193 @@
+
+            fileReader.readLine();
+
+            while (fileReader.ready()) {
```
Также необходимо будет восстанавливать максимальный id

AlexeyCheuzov commented on test/manager/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+import java.io.IOException;
+import static org.junit.jupiter.api.Assertions.*;
+
+class FileBackedTaskManagerTest {
```
Так как теперь есть система наследования, было бы хорошо использовать ее в тестах. Можно сделать параметризованный абстрактный тестовый класс, который будет хранить в себе TaskManager и содержать тесты к его методам. Далее наследовать от него,  подставлять конкретную реализацию и дополнять тесты при необходимости. Если будет не совсем понятно можно писать мне в пачку по этому вопросу.

ArtemixSled committed:
- Теперь метод loadFromFile, в классе FileBackedTaskManager, восстанавливает максимальный id
- Теперь метод loadFromFile, в классе FileBackedTaskManager, не использует метод для создания задачи, а добавляет задачи напрямую в хранилище
- Собственное исключение ManagerSaveException вынесено в отдельный класс
- Реализован в тестах абстрактный класс AbstractTaskManagerTest
